internal/controllers/addon/phase_observe_operatorresource.go

```diff
@@ -20,6 +20,45 @@ func (r *olmReconciler) observeOperatorResource(
 	addon *addonsv1alpha1.Addon,
 	csvKey client.ObjectKey,
 ) (requeueResult, error) {
+	if addon == nil {
+		return resultRetry, fmt.Errorf("current addon is nil")
+	}
+
+	installOLMCommon, err := addon.GetAddonInstallOLMCommon()
+	if err != nil {
+		return resultRetry, err
+	}
+
+	currentSub := operatorsv1alpha1.Subscription{}
+	err = r.GetSubscription(
+		ctx,
+		SubscriptionName(addon),
+		installOLMCommon.Namespace,
+		&currentSub,
+	)
+	if err != nil {
+		return resultRetry, err
+	}
+
+	currentIp := operatorsv1alpha1.InstallPlan{}
+	if currentSub.GetInstallPlanApproval() == operatorsv1alpha1.ApprovalManual {
+		err := r.GetInstallPlan(
+			ctx,
+			currentSub.Status.InstallPlanRef.Name,
+			currentSub.Status.InstallPlanRef.Namespace,
+			&currentIp,
+		)
+		if err != nil {
+			return resultRetry, err
+		}
+
+		if currentIp.Status.Phase == operatorsv1alpha1.InstallPlanPhaseRequiresApproval {
+			reportInstallPlanPending(addon)
+			// CSV will not be available at this stage therefore retry
+			return resultRetry, nil
```

Comment:
We should not be retrying here since an install plan will not become approved via eventual consistency. We should return early, but we won't need to reconcile again until the `InstallPlan` is approved. Since we aren't watching `InstallPlans` that would probably be triggered by the creation of the new CSV.